### PR TITLE
[Common Lisp] Extract Concepts from Word Count

### LINKED ISSUE #496: [Common Lisp] Extract Concepts from v2 exercise: word-count
Extract the list of concepts required to solve the v2 common-lisp exercise: `word-count`. For each concept, describe why it is required to solve the exercise. Keep in mind that an exercise can have multiple approaches (you can find community solutions here: `https://exercism.io/tracks/common-lisp/exercises/word-count/solutions`). The goal of this task is to help determine the total list of concepts that make up `word-count`.

The list of concepts should be stored in the following document: `languages/common-lisp/reference/exercise-concepts/word-count.md`. [Use this link to create a PR to add this document][pr-link].

[pr-link]: ../new/master?filename=languages/common-lisp/reference/exercise-concepts/word-count.md

### PR DESCRIPTION
**Pull request by TheLostLambda**
Closes #496. Would love some feedback!

### CODE REVIEW

**Comment by TheLostLambda** [comment]
Additionally, should `Quicklisp` be it's own concept? I feel that it could be, but it probably wouldn't have too much too it.

File: languages/common-lisp/reference/exercise-concepts/word-count.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# General Concepts of Word Count
```
Comment (verdammelt) [inline languages/common-lisp/reference/exercise-concepts/word-count.md]: Reviewing this I see I need to do a better job in my PR! This is good and detailed!

File: languages/common-lisp/reference/exercise-concepts/word-count.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  functions. Defined with `flet` or `labels`
+- Higher Order Functions: The use of various functions like `remove-if` and
+  `map` is common
+- Anonymous Functions: Some students may pass a `lambda` to functions like `map`
+- Lists: This exercise requires a result given as an `alist`
+- Boolean Logic: Compositions of `and` and `or`. Helpfully short-circuiting
+
+# String -> Words
+## Approach: Library Import
+- Regular Expressions: Needed if the student is using the `cl-ppcre`
```
Comment (verdammelt) [inline languages/common-lisp/reference/exercise-concepts/word-count.md]: I still hold the stance of *not* having students use 3rd party libraries.

File: languages/common-lisp/reference/exercise-concepts/word-count.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+## Approach: Home Rolled
+- Enumeration: Common Lisp's `loop` macro makes it possible to loop through the
```
Comment (verdammelt) [inline languages/common-lisp/reference/exercise-concepts/word-count.md]: What about `do`/`do*`? Perhaps that is not an issue of concepts but for the analyzer/recognizer to know that `do` /or/ `loop` can be used.

Thoughts?

**Comment by verdammelt** [comment]
I don't see QuickLisp as a concept - unless other languages have concepts about 'common tools' or 'environments' or 'dependency management'.

i..e Does the Ruby track have a concept about bundler or the JavaScript track have a concept about npm or yarn? 

That is what QuickLisp is to me. It is on par with bundler, npm, yarn, maven, ant. It is not part of the language, it is a very useful but not required part of the ecosystem.

File: languages/common-lisp/reference/exercise-concepts/word-count.md (hunk 2 above)
Comment (TheLostLambda) [reply languages/common-lisp/reference/exercise-concepts/word-count.md]: I also somewhat agree with that, but looking through the solutions to this exercise, many people did adopt this approach, so I'm a bit stuck on the fence personally.

I agree that we should be teaching the language, but there are also some places (perhaps we could isolate them under a super-concept of "Popular Libraries" that show off some common add-ons (e.g. UIOP, Alexandria maybe, etc)?

**Comment by TheLostLambda** [comment]
Hmm, this exists: https://github.com/exercism/v3/blob/master/reference/tooling/README.md

But I'd have to look into if other tracks include tooling as a concept or not. That might be something to ask the other tracks about.

File: languages/common-lisp/reference/exercise-concepts/word-count.md (hunk 3 above)
Comment (TheLostLambda) [reply languages/common-lisp/reference/exercise-concepts/word-count.md]: That's a good point! I didn't see too many solutions using `do`, but I should definitely include it here. I'll make that change tomorrow hopefully

**Comment by TheLostLambda** [comment]
I've downgraded Regex as a concept. I think there might be a place for it in some "common libraries" section, but I agree that it's not a main language concept.

**Pull request merged**